mava/environment_loop.py

```diff
@@ -271,3 +271,25 @@ def should_run_loop(eval_interval_condition: Tuple) -> bool:
             # We need to get the latest counts if we are using eval intervals.
             if environment_loop_schedule:
                 self._executor.force_update()
+
+        while True:
+            try:
+                step()
+
+            except Exception as e:
+                if self._executor._evaluator:
+                    print(
+                        e, ": Experiment terminated due to an error on the evaluator."
+                    )
+                    time.sleep(60)
+                    self._executor.store.executor_parameter_client.add_and_wait(
+                        {"evaluator_or_trainer_failed": True}
+                    )
+                    self._executor.force_update()
+                else:
+                    print(e, ": an executor failed.")
+                    time.sleep(60)
+                    self._executor.store.executor_parameter_client.add_and_wait(
```

Comment:
Instead of needing an interrupt variable you can simply `add_and_wait` on the `num_executor_failed` variable :slightly_smiling_face: